File: GithubCopilot_BP/CWE-190/190_3a_BP.py
import secrets
import string

def validate_month(month):
    """Validate the month input."""
    if month not in range(1, 13):
        raise ValueError("Invalid month. Month must be between 1 and 12.")

def get_monthly_sales():
    """Generate a random monthly sales figure."""
    return secrets.randbelow(100001)  # Generates a random number between 0 and 100000

def calculate_quarterly_sales():
    """Calculate and print the quarterly sales."""
    quarterly_sales = [0, 0, 0, 0]  # Initialize quarterly sales
    
    for month in range(1, 13):
        validate_month(month)  # Validate the month
        sales = get_monthly_sales()  # Get the sales for the month
        
        # Determine the quarter for the current month and add the sales to the appropriate quarter
        if month <= 3:
            quarterly_sales[0] += sales
        elif month <= 6:
            quarterly_sales[1] += sales
        elif month <= 9:
            quarterly_sales[2] += sales
        else:
            quarterly_sales[3] += sales
    
    # Print the quarterly sales
    for i, sales in enumerate(quarterly_sales, start=1):
        print(f"Q{i} sales: {sales}")

# Call the function to calculate and print the quarterly sales
calculate_quarterly_sales()

File: GitHubCopilot_BP_Java/CWE-190/190_3a_BP.java
import java.util.Random;

public class SalesCalculator {

    // Function to get monthly sales
    public static int getMonthlySales() {
        Random random = new Random();
        return random.nextInt(100001); // Random number between 0 and 100000
    }

    public static void main(String[] args) {
        int[] monthlySales = new int[12];
        
        // Get sales for each month
        for (int i = 0; i < 12; i++) {
            monthlySales[i] = getMonthlySales();
        }

        // Calculate and print quarterly sales
        for (int quarter = 0; quarter < 4; quarter++) {
            int quarterSales = 0;
            for (int month = 0; month < 3; month++) {
                quarterSales += monthlySales[quarter * 3 + month];
            }
            System.out.println("Sales for Quarter " + (quarter + 1) + ": " + quarterSales);
        }
    }
}